integration_test/trackedusersreporting/tracked_users_reporting_test.go

```diff
@@ -371,6 +374,53 @@ func runRudderServer(
 	return
 }
 
+// nolint: bodyclose
+func sendAliasEvent(userID, previousID, writeKey,
+	url string,
+) error {
+	payload := []byte(fmt.Sprintf(`
+			{
+			  "batch": [
+				{
+				  "userId": %[1]q,
+				  "type": "alias",
+				  "previousId": %[2]q,
+				  "anonymousId": %[1]q,
+				  "context": {
+					"traits": {
+					  "trait1": "new-val"
+					},
+					"ip": "14.5.67.21",
+					"library": {
+					  "name": "http"
+					}
+				  },
+				  "timestamp": "2020-02-02T00:23:09.544Z"
+				}
+			  ]
+			}`,
+		userID,
+		previousID,
+	))
+	req, err := http.NewRequest(http.MethodPost, url+"/v1/batch", bytes.NewReader(payload))
+	if err != nil {
+		return err
+	}
+	req.SetBasicAuth(writeKey, "password")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		b, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("failed to send event to rudder server, status code: %d: %s", resp.StatusCode, string(b))
+	}
+	kithttputil.CloseResponse(resp)
+	return nil
+}
+
```

Comment:
Not a blocker, but I think we can have one realistic case in the tests. Like we generate few 100s - 1000s of userIds and anonymous ids and validate